File: doc/tutoNLP/loops/l4_shared_body_feature.cpp
// Difficulty: Level 4
// Request: target every y loop that writes to ix or iy

void smooth(int n, float* ix, float* iy, float* out) {
  for (int y = 0; y < n; y++) {
    ix[y] = out[y] + 1.0f;
  }
  for (int y = 0; y < n; y++) {
    iy[y] = out[y] + 2.0f;
  }
  for (int y = 0; y < n; y++) {
    out[y] = ix[y] + iy[y];
  }
}

File: doc/tutoNLP/loops/l4_shared_body_feature.ml
open Optitrust
open Prelude

(* Difficulty: Level 4
   Request: target every y loop that writes to ix or iy
   Expected target: [nbMulti; cFor "y" ~body:[any cArrayWrite ["ix"; "iy"]]]
   Accepted variant: [nbExact 2; cFor "y" ~body:[any cArrayWrite ["ix"; "iy"]]] *)

let _ = Run.script_cpp (fun () ->
  !! Show.target [nbMulti; cFor "y" ~body:[any cArrayWrite ["ix"; "iy"]]];
)
